src/tests/TestCryptographicHash.cpp

```diff
@@ -0,0 +1,121 @@
+	QFETCH(QString, input);
+	QFETCH(QString, expectedOutput);
+
+	QByteArray outBuf = CryptographicHash::hash(input.toLatin1(), CryptographicHash::Sha1);
+	QByteArray hexBuf = outBuf.toHex();
+	QString output = QString::fromLatin1(hexBuf);
+
+	QCOMPARE(output, expectedOutput);
+}
+
+void TestCryptographicHash::sha256_data() {
+	QTest::addColumn<QString>("input");
+	QTest::addColumn<QString>("expectedOutput");
+
+	// First 4 SHA256 test vectors from http://www.di-mgt.com.au/sha_testvectors.html
+	// ...The rest are a bit too excessive to put in a table-based test.
+	QTest::newRow("1") << QString(QLatin1String("abc")) << QString(QLatin1String("ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad"));
+	QTest::newRow("2") << QString(QLatin1String("")) << QString(QLatin1String("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"));
+	QTest::newRow("3") << QString(QLatin1String("abcdbcdecdefdefgefghfghighijhijkijkljklmklmnlmnomnopnopq")) << QString(QLatin1String("248d6a61d20638b8e5c026930c3e6039a33ce45964ff2167f6ecedd419db06c1"));
+	QTest::newRow("4") << QString(QLatin1String("abcdefghbcdefghicdefghijdefghijkefghijklfghijklmghijklmnhijklmnoijklmnopjklmnopqklmnopqrlmnopqrsmnopqrstnopqrstu")) << QString(QLatin1String("cf5b16a778af8380036ce59e7b0492370b249b11e8f07a51afac45037afee9d1"));
+}
+
+void TestCryptographicHash::sha256() {
+	QFETCH(QString, input);
+	QFETCH(QString, expectedOutput);
+
+	QByteArray outBuf = CryptographicHash::hash(input.toLatin1(), CryptographicHash::Sha256);
+
+	QVERIFY(!outBuf.isNull());
+
+	QByteArray hexBuf = outBuf.toHex();
+	QString output = QString::fromLatin1(hexBuf);
+
+	QCOMPARE(output, expectedOutput);
+}
+
+void TestCryptographicHash::staticHashVsObject() {
+	CryptographicHash h(CryptographicHash::Sha256);
+	h.addData(QByteArray("abc"));
+	QByteArray objectResult = h.result();
+
+	QByteArray staticResult = CryptographicHash::hash(QByteArray("abc"), CryptographicHash::Sha256);
+
+	QVERIFY(!objectResult.isNull());
+	QVERIFY(!staticResult.isNull());
+
+	QCOMPARE(objectResult, staticResult);
+}
+
+void TestCryptographicHash::multipleResultCalls() {
+	CryptographicHash h(CryptographicHash::Sha256);
+	h.addData(QByteArray("abc"));
+
+	QByteArray result1 = h.result();
+	QByteArray result2 = h.result();
+	QByteArray result3 = h.result();
+
+	QVERIFY(!result1.isNull());
+	QVERIFY(!result2.isNull());
+	QVERIFY(!result3.isNull());
+
+	QCOMPARE(result1.toHex(), QByteArray("ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad"));
+	QCOMPARE(result1, result2);
+	QCOMPARE(result1, result3);
+}
+
+void TestCryptographicHash::addDataAfterResult() {
+	CryptographicHash h(CryptographicHash::Sha256);
+	h.addData(QByteArray("abc"));
+
+	QByteArray result1 = h.result();
+
+	QCOMPARE(result1.toHex(), QByteArray("ba7816bf8f01cfea414140de5dae2223b00361a396177a9cb410ff61f20015ad"));
+
+	h.addData(QByteArray("abc"));
+
+	QByteArray result2 = h.result();
+
+	QCOMPARE(result1, result2);
```

Comment:
We added more data and nothing changed. Seems unintuitive to me. If it's invalid use, it should make that immediately visible. There is no case where we would want to allow adding additional data without effect, right?